src/python/WMCore/MicroService/Unified/MSOutput.py

```diff
@@ -677,57 +714,69 @@ def docInfoUpdate(self, msOutDoc, pipeLine=None):
                     destination.add('T2_CH_CERN')
                 if "MinimumBias" in dsn and "SiStripCalMinBias" in procString and dataTier != "ALCARECO":
                     destination.add('T2_CH_CERN')
-                dMap = {'datasets': [dataset],
-                        'destination': list(destination)}
-                destinationOutputMap.append(dMap)
-                wflowDstSet |= destination
+            else:
+                # FIXME:
+                #    Here we need to use the already created campaignMap for
+                #    building the destinationOutputMap for nonRelVal workflows.
+                #    For the time being it is a fallback to all T1_* and all T2_*.
+                #    Once we migrate to Rucio we should change those defaults to
+                #    whatever is the format in Rucio (eg. referring a subscription
+                #    rule like: "store it at a good site" or "Store in the USA" etc.)
+                destination.add('T1_*')
+                destination.add('T2_*')
+
+            dMap = {'datasets': [dataset],
+                    'destination': list(destination)}
+            destinationOutputMap.append(dMap)
+            wflowDstSet |= destination
+
+        # here we try to aggregate the destination map per destination
+        aggDstMap = []
+
+        # populate the first element in the aggregated list
+        if len(destinationOutputMap) != 0:
+            aggDstMap.append(destinationOutputMap.pop())
 
-            # here we try to aggregate the destination map per destination
-            aggDstMap = []
+        # feed the rest
+        while len(destinationOutputMap) != 0:
```

Comment:
I think this can be addressed in a much cleaner way with dictionary. The key will take care of the dataset aggregation by site(s).

Reply:
I am not sure about that. The campaign map is also a list, having them in the same format is good. Also changing the format here will require a change of the database document format and a whole new testing and validation of how we build the map and how we use it etc. etc.. Not that it is quite a big test but will require time we do not have it right now.

Reply:
Dict would be a temporary data structure. Just so you can aggregate datasets by site. In the end you would then need to convert that dict to a list of dicts, as defined in the document schema.

However, this destination output map really does not belong to MSOutput and it should be moved under the msOutTemplate module. That class should contain all the logic for a workflow <--> msOutDoc (including the RelVal logic. Can you please move it there and get unit tests created?